code/modules/projectiles/guns/energy/kinetic_accelerator.dm

```diff
@@ -482,7 +482,7 @@
 			R.damage_multiplier = modifier
 			R.burst()
 			return
-		new /obj/effect/temp_visual/resonance(target_turf, K.firer, null, 30)
+		new /obj/effect/temp_visual/resonance(target_turf, K.firer, null, 2, 100) //manual detonate mode and will NOT spread
```

Comment:
Yeah, that's pretty annoying. Especially the duplicated `locate` stuff that is just above this. It's not your code though and you've already refactored a bunch, so let's focus on the mode param instead:

Some options:
1) Move the RESONATOR_MODE_* definitions to somewhere such as `code/__DEFINES/resonator.dm` and use `RESONATOR_MODE_MANUAL`  instead of `2`
2) Add a helper global-proc for creating this effect? Kinda lame unless you _also_ move the RESONATOR_MODE_* stuff, but this would let you clean up the duplicated `locate` logic.
3) Add a new typepath (like `/obj/effect/temp_visual/resonance/kinetic_accelerator_upgrade`) which has an Initialize method that only needs the first param. I don't really like this one:
    ```dm
        /obj/effect/temp_visual/resonance/kinetic_accelerator_upgrade/Initialize(mapload, set_creator)
            . = ..(mapload, set_creator, null, RESONATOR_MODE_MANUAL, 
    ```

I'd go for 1 unless you are really interested in cleaning up the ka resonance mod.

Reply:
definitely going for 1. Though, cleaning up that KA file might be a good thing to do later on (in a separate PR).